### ISSUE TITLE
users-setup and upgrade procedures missing gnt-wconfd and gnt-metad users/groups.

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1032]
Originally reported of Google Code with ID 976.

```
Ganeti 2.12, debian sid package on a mostly debian testing sytem

gnt-cluster (ganeti v2.12.0) 2.12.0
Software version: 2.12.0
Internode protocol: 2120000
Configuration format: 2120000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.12.0
hspace (ganeti) version v2.12.0
compiled with ghc 7.6
running on linux x86_64


Upgrading to Ganeti 2.12 produces the problem (during upgrade).


If I am reading the error message correctly, Ganeti 2.12 does not create a required group gnt-wconfd.  This appears to be an upstream bug.

https://code.google.com/p/ganeti/issues/detail?id=952
https://code.google.com/p/ganeti/issues/detail?id=953

# On /all/ nodes

#
# NOTE: Upstream does not yet have changes in the makefile.am
#  to create the users-setup with any gnt-wconfd / gnt-metad u/g
# I have /guessed/ at some values that seem to make sense and work for me.
#

addgroup --system gnt-wconfd
addgroup --system gnt-metad
adduser --system --no-create-home --ingroup gnt-wconfd --ingroup gnt-masterd --ingroup gnt-metad gnt-wconfd
adduser --system --no-create-home --ingroup gnt-metad gnt-metad

# Then on the master node:

gnt-cluster upgrade --to 2.12

# You may also need to restart the ganeti daemons on all nodes one additional time.



--

gnt-cluster upgrade --to 2.12
…
    _TestLoadingConfigFile()
  File "/usr/lib/ganeti/tools/cfgupgrade", line 678, in _TestLoadingConfigFile
    vrfy = cfg.VerifyConfig()
  File "/usr/share/ganeti/2.12/ganeti/config.py", line 115, in sync_function
    with args[0].GetConfigManager(shared):
  File "/usr/share/ganeti/2.12/ganeti/config.py", line 243, in __enter__
    self._config_writer._OpenConfig(self._shared) # pylint: disable=W0212
  File "/usr/share/ganeti/2.12/ganeti/config.py", line 2797, in _OpenConfig
    self._UpgradeConfig(saveafter=True)
  File "/usr/share/ganeti/2.12/ganeti/config.py", line 2878, in _UpgradeConfig
    self._WriteConfig()
  File "/usr/share/ganeti/2.12/ganeti/config.py", line 2900, in _WriteConfig
    getents = self._getents()
  File "/usr/share/ganeti/2.12/ganeti/runtime.py", line 221, in GetEnts
    _priv = resolver() # pylint: disable=W0621
  File "/usr/share/ganeti/2.12/ganeti/runtime.py", line 114, in __init__
    self.wconfd_gid = GetGid(constants.WCONFD_GROUP, _getgrnam)
  File "/usr/share/ganeti/2.12/ganeti/runtime.py", line 81, in GetGid
    raise errors.ConfigurationError("Group '%s' not found (%s)" % (group, err))
ganeti.errors.ConfigurationError: Group 'gnt-wconfd' not found ('getgrnam(): name not found: gnt-wconfd')

root@h-2:~# /usr/lib/ganeti/tools/users-setup
This will do the following changes
+ Will add group gnt-admin
+ Will add group gnt-confd
+ Will add group gnt-daemons
+ Will add group gnt-luxid
+ Will add group gnt-masterd
+ Will add group gnt-rapi
+ Will add user gnt-confd with primary group gnt-confd
+ Will add user gnt-masterd with primary group gnt-confd
+ Will add user gnt-masterd with primary group gnt-luxid
+ Will add user gnt-masterd with primary group gnt-masterd
+ Will add user gnt-rapi with primary group gnt-rapi
+ Will add user gnt-confd to group gnt-daemons
+ Will add user gnt-masterd to group gnt-admin
+ Will add user gnt-masterd to group gnt-confd
+ Will add user gnt-masterd to group gnt-daemons
+ Will add user gnt-masterd to group gnt-masterd
+ Will add user gnt-rapi to group gnt-admin
+ Will add user gnt-rapi to group gnt-daemons

OK? (y/n)
y
addgroup: The group `gnt-admin' already exists as a system group. Exiting.
addgroup: The group `gnt-confd' already exists as a system group. Exiting.
addgroup: The group `gnt-daemons' already exists as a system group. Exiting.
addgroup: The group `gnt-luxid' already exists as a system group. Exiting.
addgroup: The group `gnt-masterd' already exists as a system group. Exiting.
addgroup: The group `gnt-rapi' already exists as a system group. Exiting.
The system user `gnt-confd' already exists. Exiting.
The system user `gnt-masterd' already exists. Exiting.
The system user `gnt-masterd' already exists. Exiting.
The system user `gnt-masterd' already exists. Exiting.
The system user `gnt-rapi' already exists. Exiting.
The user `gnt-confd' is already a member of `gnt-daemons'.
The user `gnt-masterd' is already a member of `gnt-admin'.
The user `gnt-masterd' is already a member of `gnt-confd'.
The user `gnt-masterd' is already a member of `gnt-daemons'.
The user `gnt-masterd' is already a member of `gnt-masterd'.
The user `gnt-rapi' is already a member of `gnt-admin'.
The user `gnt-rapi' is already a member of `gnt-daemons'.

```
Originally added on 2014-10-17 18:51:31 +0000 UTC.

Mentioned Issue with GoogleCode ID 952 has been migrated to GitHub with ID #1008.
Mentioned Issue with GoogleCode ID 953 has been migrated to GitHub with ID #1009.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Assigning to pudlak as it is wconfd-related.
```
Originally added on 2014-10-21 08:16:22 +0000 UTC.

Added Labels: Type-Defect Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-10-21 08:42:22 +0000 UTC.

Added to Milestone: Release2.12
Removed from Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
The actual problem was, that the daemons were trying to run in
that group in the first place. Fixed in 2.12 (will be part of
2.12.1, already fixed locally in the debian package as 2.12.0-3)
in commit cd81fbb
```
Originally added on 2014-10-21 08:44:05 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-01-15 11:04:29 +0000 UTC.

Changed State: Released
